### PR TITLE
2 layer implementation with enhancements

### PR DESCRIPTION
**Pull request by jayalakshmi-sasidharan**
Data augmentation, Weighted decay, and Class imbalance with weighted loss function implemented.

### CODE REVIEW

**Review by ankita-m-john — APPROVED** [review]
Good work with hyperparameter tuning.
